trie/sync.go

```diff
@@ -425,6 +445,30 @@ func (s *Sync) children(req *nodeRequest, object node) ([]*nodeRequest, error) {
 			node: node.Val,
 			path: append(append([]byte(nil), req.path...), key...),
 		}}
+		// Mark all internal nodes between shortNode and its **in disk**
+		// child as invalid. This is essential in the case of path mode
+		// scheme; otherwise, state healing might overwrite existing child
+		// nodes silently while leaving a dangling parent node within the
+		// range of this internal path on disk. This would break the
+		// guarantee for state healing.
+		//
+		// This step is only necessary for path mode, as there is no deletion
+		// in hash mode at all.
+		if _, ok := node.Val.(hashNode); ok && s.scheme == rawdb.PathScheme {
+			owner, inner := ResolvePath(req.path)
+			for i := 1; i < len(key); i++ {
+				// Theoretically, it's necessary to check for the presence before
+				// blindly caching deletion commands. However, due to the fact that
+				// Pebble doesn't use a bloom filter to enhance read performance
+				// for non-existent items, this check would significantly slow down
+				// overall performance. FIX IT(rjl493456442)
+				if rawdb.HasTrieNodeInPath(s.database, owner, append(inner, key[:i]...)) {
```

Comment:
```
// Theoretically, it's necessary to check for the presence before
// blindly caching deletion command
...
if rawdb.HasTrieNodeInPath (...
```
You _do_ check for presence, before deletion, no?

Reply:
It _should_ be possible to make this faster with some range-delete, alternatively use a range-iterator + delete. Because the layout of the keys are incremental, it should be reasonably efficient

Reply:
Yes, i just want to see the performance impact, but looks like no big difference..

Reply:
Range deletions makes a bit more assumptions about the db layout than I'm comfortable with. We can definitely test the performance and see how often this happens to consider if it's worth it. But if there's a way, I'd prefer to be precise vs range delete.